### PR TITLE
NewClient panics if http.client is nil and LINODE_CA is set

### PR DESCRIPTION
**Pull request by kokes**
I tried a new linodego with a custom CA and a nil HTTP client and encountered a panic. Here's a minimal reproducer (that I now include in the test suite):

```
func TestClient_CustomRootCAWithoutCustomRoundTripper(t *testing.T) {
	caFile, err := os.CreateTemp(t.TempDir(), "linodego_test_ca_*")
	if err != nil {
		t.Fatalf("Failed to create temp ca file: %s", err)
	}
	defer os.Remove(caFile.Name())

	t.Setenv(APIHostCert, caFile.Name())

        NewClient(nil)
}
```

Once I run this test, I get:

```
$ go test -run=TestClient_CustomRootCAWithoutCustomRoundTripper ./...
?   	github.com/linode/linodego/cmd	[no test files]
?   	github.com/linode/linodego/internal/parseabletime	[no test files]
?   	github.com/linode/linodego/internal/testutil	[no test files]
--- FAIL: TestClient_CustomRootCAWithoutCustomRoundTripper (0.00s)
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x0 pc=0x102dbb3d8]

goroutine 22 [running]:
testing.tRunner.func1.2({0x102fb5dc0, 0x1034115e0})
	/opt/homebrew/Cellar/go/1.23.3/libexec/src/testing/testing.go:1632 +0x1bc
testing.tRunner.func1()
	/opt/homebrew/Cellar/go/1.23.3/libexec/src/testing/testing.go:1635 +0x334
panic({0x102fb5dc0?, 0x1034115e0?})
	/opt/homebrew/Cellar/go/1.23.3/libexec/src/runtime/panic.go:785 +0x124
github.com/linode/linodego.NewClient(0x0)
	/Users/okokes/git/linode/public/linodego/client.go:741 +0x208
github.com/linode/linodego.TestClient_CustomRootCAWithoutCustomRoundTripper(0x140000c6b60)
	/Users/okokes/git/linode/public/linodego/client_test.go:589 +0xdc
testing.tRunner(0x140000c6b60, 0x10306a230)
	/opt/homebrew/Cellar/go/1.23.3/libexec/src/testing/testing.go:1690 +0xe4
created by testing.(*T).Run in goroutine 1
	/opt/homebrew/Cellar/go/1.23.3/libexec/src/testing/testing.go:1743 +0x314
FAIL	github.com/linode/linodego	1.116s
ok  	github.com/linode/linodego/internal/duration	0.713s [no tests to run]
FAIL
```

This is a followup to #593 - in this PR this LINODE_CA support got fixed, but not in the case of `hc == nil`, the constructor would panic upon `hc.Transport`.

### CODE REVIEW

**Review by zliang-akamai — APPROVED** [review]
LGTM!

File: client.go [hunk 1]
```diff
@@ -881,8 +881,11 @@ func generateListCacheURL(endpoint string, opts *ListOptions) (string, error) {
 	return fmt.Sprintf("%s:%s", endpoint, hashedOpts), nil
 }
 
-func isCustomTransport(transport http.RoundTripper) bool {
-	if transport != http.DefaultTransport.(*http.Transport) {
+func isCustomTransport(hc *http.Client) bool {
```
Comment (zliang-akamai) [inline client.go:884]:
```suggestion
func hasCustomTransport(client *http.Client) bool {
```

How about this new function name? Technically it's no longer checking hc but the http client

**Comment by jriddle-linode** [comment]
@kokes looks like it needs a `make format`

**Comment by kokes** [comment]
I committed from my phone and didn't realise your inline suggestion only fixed the function definition, not the call itself, updated that in a subsequent commit.

**Review by jriddle-linode — APPROVED** [review]
LGTM

**Pull request merged**